### ISSUE TITLE
Structures can become larger/smaller than normal graphically

### ISSUE THREAD

**Issue by nvs** [nvs/gem#81]
Seems to be related to alt-tabbing.

**Comment by nvs** [comment]
It might be possible to deal with this by ensuring the model size is correct, or doing a reset of the size periodically.

**Comment by nvs** [comment]
A bit of discussion and thinking leads me to believe this is related to upgrading a structure. It typically shows up on towers such as Stars, Silvers, and Blood Stones. It is very likely that using `SetUnitSize ()` after construction/upgrading could resolve the problem.

As far as testing, zooming through a replay seems to be enough to see the local graphical issue. Need to get a baseline for how common it is, then I should be able to gauge whether that function actually fixes the issue.

**Comment by nvs** [comment]
This bug appearing in-game has been rare lately. I cannot remember the last time I saw it since 1.30.1 appeared. Going to wait on addressing this, as the ability to get a unit's scale is necessary, and most have unique values. Should be easily handled in 1.6.0.

**Comment by nvs** [comment]
This is blocked by #23.

**Comment by nvs** [comment]
This primarily affects only replays now. As such, it is a low priority bug.

**Comment by nvs** [comment]
The new natives in 1.31 may allow accessing the scaling value. This would make it quite easy to address this graphical bug.

**Comment by nvs** [comment]
Work done in 545b6e490d7e5cc95a02d79f9b655677fdb18420 may have addressed this bug. Essentially, upon an upgrade finishing the scale of the unit would be wrong. This has not really been an issue during gameplay since either 1.29 or 1.30. It has, however, been a problem in replays.

**Comment by nvs** [comment]
I believe this was resolved due to work in 1.6, specifically in 1.6.7. And then finally resolved in 1.6.8, due to an issue introduced in 1.6.7.
